identity-service/src/routes/relay.js

```diff
@@ -1,27 +1,47 @@
 const crypto = require('crypto')
 
-const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError } = require('../apiHelpers')
+const { handleResponse, successResponse, errorResponseBadRequest, errorResponseForbidden, errorResponseServerError } = require('../apiHelpers')
 const txRelay = require('../relay/txRelay')
 const captchaMiddleware = require('../captchaMiddleware')
 const { detectAbuse } = require('../utils/antiAbuse')
 const { getFeatureFlag, FEATURE_FLAGS } = require('../featureFlag')
 
+const blockRelayAbuseErrorCodes = new Set(['3', '8', '9', '10', '11'])
+
 module.exports = function (app) {
   // TODO(roneilr): authenticate that user controls senderAddress somehow, potentially validate that
   // method sig has come from sender address as well
```

Comment:
nit: we could probably get rid of this todo since we have the auth middleware TODO (let's make a Linear for that one?)